components/permissions/google_sign_in/google_sign_in_permission_throttle.cc

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/permissions/google_sign_in/google_sign_in_permission_throttle.h"
+
+#include <utility>
+#include <vector>
+
+#include "base/functional/bind.h"
+#include "brave/components/permissions/google_sign_in/google_sign_in_permission_util.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/navigation_entry.h"
+#include "content/public/browser/permission_controller_delegate.h"
+#include "content/public/browser/web_contents.h"
+#include "extensions/common/url_pattern.h"
```

Comment:
`extensions` cannot be used in `components`. Please replace `URLPattern` with
`components/content_settings/core/common/content_settings_pattern.h`
or
`components/url_pattern_index/url_pattern.h`